infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -139,11 +143,15 @@ public LogUnitServer(ServerContext serverContext) {
                 .maximumWeight(maxCacheSize)
                 .removalListener(this::handleEviction)
                 .writer(batchWriter)
+                .recordStats()
                 .build(this::handleRetrieval);
 
         // Trim map is set to empty on start
         // TODO: persist trim map - this is optional since trim is just a hint.
         trimMap = new ConcurrentHashMap<>();
+
+        MetricRegistry metrics = serverContext.getMetrics();
+        MetricsUtils.addCacheGauges(metrics, metricsPrefix + "cache.", dataCache);
```

Comment:
The overhead of metric objects that are in the registry but never used is quite small.  If that small overhead is a noticeable problem, then we can address it.  If we change things so that metrics are added to the registry only when collection is enabled dynamically, then the annotation-based metrics will need new closures to encapsulate timer references that weren't needed at startup time.  We'd have to run something like the equivalent of `handler = CorfuMsgHandler().generateHandlers()` for each server object when metrics collection status flips from disabled->enabled.